Latest Singular fails Darwin cross compilation

@hannes14 pointed out to me on June 4 that SingularBuilder fails to build the Darwin / macOS version of Singular as in commit https://github.com/Singular/Sources/commit/ae2e290177fd5de77a3afac81e53b9da8977da9a (BTW I *assume* this is Singular version 4.1.3, but I am not sure as there is no tag in the repository, see also https://github.com/Singular/Sources/issues/998 )

The issue can be seen in the [Travis logs](https://travis-ci.org/github/oscar-system/SingularBuilder/builds), e.g. [this one](https://travis-ci.org/github/oscar-system/SingularBuilder/builds/694807937):
```
[21:12:21]   CXXLD    TSingular
[21:12:21]   CXXLD    libparse
[21:12:22]   CXXLD    libSingular.la
[21:12:22] ld: warning: ignoring file ../kernel/.libs/libkernel.a, file was built for archive which is not the architecture being linked (x86_64): ../kernel/.libs/libkernel.a
[21:12:22]   CXXLD    Singular
[21:12:23] Undefined symbols for architecture x86_64:
[21:12:23]   "_fe_fgets_dummy", referenced from:
[21:12:23]       _main in tesths.o
[21:12:23]   "_fe_fgets_stdin", referenced from:
[21:12:23]       _main in tesths.o
```

One can reproduce this locally on a Linux system via the following command, executed in the `SingularBuilder` directory:
```
julia --project=. --color=yes -e 'using Pkg; pkg"add BinaryProvider"; pkg"add BinaryBuilder#af0b4853b2b95b1c411f06f834dcbf461bb50264"; Pkg.build()'
julia --project=. --color=yes build_tarballs.jl --verbose
```

The issue is really in Singular and not in Julia, as (a) we hardcode the BinaryBuilder.jl version and Travis fixes the build environment for us, and (b) PR #4 shows that the previous Singular commit we used still build fine.

But that means we have a "good" and a "bad" revision in the Singular repository, and we can run a `git bisect` to figure out when things started to break. I tried this, however, in the problematic range other build errors appear (some compilation units fail due to `include "gmp.h"` not being able to locate `gmp.h`; I tried to skip these, but there are just too many commits were this appears).

Once this is fixed, I strongly suggest that the above steps (adjusted to target the latest Singular commit) are executed as part of the Singular CI to make sure we catch similar issues sooner (i.e., before and not *after* Singular releases). I see several options for that, not mutually exclusive:
1. Assuming @hannes14 use a CI server for Singular in Kaiserslautern, that server could do it (does such a server exist? How can I reach it)
2. We could setup Travis or GitHub Actions or so on the Singular repository and one of the jobs it runs could be the above
3. We add a Travis or GitHub Actions cron job to the `SingularBuilder` repository which does this
4. @rbehrends adds such a test to the OSCAR Jenkins.

CC @fieker @wbhart

--- a/build_tarballs.jl
+++ b/build_tarballs.jl
@@ -19,6 +19,7 @@ export HOSTCFLAGS=-I$prefix/include
 export CFLAGS=-I$prefix/include
 export LDFLAGS=-Wl,-rpath,$prefix/lib
 export LD_LIBRARY_PATH=$target/lib:$LD_LIBRARY_PATH
+export AR=/opt/$target/bin/$target-ar
 if [ $target = "x86_64-linux-gnu" ]; then
   mkdir -p $prefix/x86_64-linux-gnu/lib/../lib64
   cp /opt/x86_64-linux-gnu/x86_64-linux-gnu/lib64/libstdc++.la /workspace/destdir/x86_64-linux-gnu/lib/../lib64/

